### ISSUE TITLE
`torchaudio.functional.istft` is even slower than IDFT implemented by ConvTranspose1d

### ISSUE THREAD

**Issue by tuxzz** [pytorch/audio#471]
## 🐛 Bug

Current implemention of `torchaudio.functional.istft` has two `(in_channel=n_fft, out_channel=1, kernel_size=n_fft,)` identity ConvTranspose1d, and 1 ifft for each frame.
A simple real IDFT implemention using ConvTranspose1d only uses single (in_channel=2 * n_bin, out_channel=1, kernel_size=win_size,) ConvTranspose1d, which has less than 0.5x computational complexity compared to `torchaudio.functional.istft`.

In testing, `torchaudio.functional.istft` costs 3x-5x time in `forward` and `backward` compared to IDFT.

## Environment
PyTorch version: 1.4.0
Is debug build: No
CUDA used to build PyTorch: 10.2

OS: Arch Linux
GCC version: (Arch Linux 9.3.0-1) 9.3.0
CMake version: version 3.17.0

Python version: 3.8
Is CUDA available: Yes
CUDA runtime version: 10.2.89
GPU models and configuration: GPU 0: GeForce RTX 2070 SUPER
Nvidia driver version: 440.64
cuDNN version: /usr/lib/libcudnn.so.7.6.5

Versions of relevant libraries:
[pip3] numpy==1.18.1
[pip3] torch==1.4.0
[pip3] torchaudio==0.4.0
[conda] Could not collect

**Comment by tuxzz** [comment]
Possible solution: replace identity `ConvTranspose1d` with `torch.nn.function.fold`
Example:
````python
...
# y = torch.nn.functional.conv_transpose1d(ytmp, eye, stride=hop_length, padding=0)
y = F.fold(ytmp, (1, n + n_fft), (1, n_fft), stride=(1, hop_length)).squeeze(1)
...
````

**Comment by mthrok** [comment]
Hi @tuxzz 

Thank you for raising the issue and PR.

I added my benchmark [here](https://github.com/pytorch/audio/pull/472#issuecomment-603279991).
If you would like to add comment, please do so.

**Comment by vincentqb** [comment]
Closed by #472
